Disable email verification when email manually changed by idp review.

rmartinc opened the pull request:
Add option to hide the idp review button.
Closes #40446

Draft PR that just adds the ideas presented in the issue comment.

1. The `Verify existing account by Email` steps is not executed when the `username` or the `email` are modified in the `Review Profile` step. This way sending emails to random users is disabled. The PR sets a configuration option to revert to previous behavior (if you think that is not needed just let me know). When disabled only the re-authentication execution is available to link the account.
2. The PR also adds an option to remove the review profile button in `Confirm link existing account`. I managed the option to just check if the `Review Profile` step was executed before. But I preferred to add the option just in case a custom review execution is used or similar. (The same, if you think that is better not to use an option and just check if the review step has been executed just let me know.)

I have chatted a bit with @pedroigor and he thinks we will need to improve email texts too. It's a good idea. We will need an upgrading note too. But I'll do both points if the general idea in this PR is OK.

Tests added. @keycloak/core-iam take a look.

sguilhen commented:
This gets a +1 from me. I am personally not sure about the option to revert to the current behavior in point 1 above. I would rather always have the username AND email change validations always in place.

rmartinc committed:
Disable email verification when email manually changed by idp review
Closes #40446

Signed-off-by: rmartinc <rmartinc@redhat.com>

rmartinc commented:
OK, I have done the changes we have been talking about:

* There is no new options for the authenticators. The email validator never send emails if the email/username is changed during review (so previous wrong behavior is not allowed anymore). This step is only available when the username/email received from IdP is used.
* The review button is not displayed if the idp review authenticator was not successfully executed. This avoids the option too.
* The emails and the confirming page of the email verification have been changed a bit. IMHO the email was quite good and  just added a sentence saying `if you didn't initiate this process...`. The confirming page was modified more to differentiate title and body, and adding a message similar to the email for the body.
* Rebased to current main.
* A upgrading note was added as a notable change commenting the changes.

Let's see the tests but they worked OK for me in another branch.

mposolda commented:
@rmartinc @pedroigor @sguilhen Thanks for this PR and for reviewing it!
